src/core_modules/capture-core/components/Pages/Enrollment/EnrollmentPage.component.js

```diff
@@ -55,14 +57,30 @@ const EnrollmentPagePlain = ({
     const { resetEnrollmentId } = useResetEnrollmentId();
     const { resetTeiId } = useResetTeiId();
 
+    const selectProgramHandler = useCallback(
+        (id) => {
+            setProgramId(id);
+            dispatch(fetchEnrollments());
+        },
+        [dispatch, setProgramId],
+    );
+
+    const deselectProgramHandler = useCallback(
+        () => {
+            resetProgramIdAndEnrollmentContext();
+            dispatch(updateEnrollmentAccessLevel({ accessLevel: enrollmentAccessLevels.UNKNOWN_ACCESS }));
+        },
+        [dispatch, resetProgramIdAndEnrollmentContext],
+    );
```

Comment:
`useCallback` is useful when passing callbacks to optimize child components and avoid unnecessary rendering. We don't always use it because most of our components are lightweight and their re-rendering doesn't affect performance. But in this particular case, `ScopeSelector` is quite a complex component, and in my opinion, it's a good idea to try to avoid unnecessary re-renders as much as possible.

Reply:
If I'm not mistaken the only

Reply:
No, that was slightly wrong; the function bodies associated with the pointers also need to persist, which also to some degree may contribute to increased memory consumption.. Or that's probably not relevant, because the dependency function pointers will always be defined _somewhere else_ before they can be inserted in the dependency array. Hence the memory allocated to the function bodies of the functions listed the dependency array is not `useCallback`'s fault, so seems like 24 bytes is correct after all :thinking: